core/types/header_ext.go

```diff
@@ -0,0 +1,197 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"io"
+	"math/big"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/common/hexutil"
+	ethtypes "github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/rlp"
+)
+
+// HeaderExtra is a struct that contains extra fields used by Avalanche
+// in the block header.
+// This type uses HeaderSerializable to encode and decode the extra fields
+// along with the upstream type for compatibility with existing network blocks.
+type HeaderExtra struct {
+	ExtDataHash    common.Hash
+	ExtDataGasUsed *big.Int
+	BlockGasCost   *big.Int
+}
+
+func (h *HeaderExtra) EncodeRLP(eth *ethtypes.Header, writer io.Writer) error {
+	out := new(HeaderSerializable)
+
+	updateFromEth(out, eth)
+	updateFromExtras(out, h)
+
+	return rlp.Encode(writer, out)
+}
+
+func (h *HeaderExtra) DecodeRLP(eth *ethtypes.Header, stream *rlp.Stream) error {
+	in := new(HeaderSerializable)
+	if err := stream.Decode(in); err != nil {
+		return err
+	}
+
+	updateToEth(in, eth)
+	updateToExtras(in, h)
+
+	return nil
+}
+
+//nolint:stdmethods
```

Comment:
I think this is ineffective, the warning comes from gopls AfAIK and no way to disable it unfortunately (again, AFAIK)